### PR TITLE
Fix SSL support between master and node(s)

### PR DESCRIPTION
**Pull request by jwhonce**

### CODE REVIEW

**Comment by jwhonce** [comment]
@twiest Please review
@detiber @akram FYI

File: roles/openshift_node/tasks/main.yml [hunk 1]
```diff
@@ -37,7 +37,8 @@
 - name: Save firewalld port for OpenShift
   firewalld: port=10250/tcp permanent=true state=enabled
 
+  # Always bounce service to pick up new credentials
 - name: Enable OpenShift
-  service: name=openshift-node enabled=yes state=started
+  service: name=openshift-node enabled=yes state=restarted
```
Comment (twiest) [inline roles/openshift_node/tasks/main.yml]: I don't think we want restarted here. If you're trying to bounce it to pick up new creds, we should use the handler when the creds are updated.

From the ansible service module doc  "restarted will always bounce the service."

http://docs.ansible.com/service_module.html

Comment (detiber) [inline roles/openshift_node/tasks/main.yml]: The handler is already being invoked when file is changed.

**Comment by detiber** [comment]
@jwhonce you'll also want to add something like this: https://github.com/detiber/openshift-online-ansible/blob/enterprise2/roles/base_os/tasks/main.yaml#L14-L20
Otherwise, the cli commands are broken by default with https enabled.

File: roles/openshift_node/tasks/main.yml (hunk 1 above)
Comment (akram) [inline roles/openshift_node/tasks/main.yml]: You could also factorise the declaration of the firewalld in a more concise syntax like this:

```
- name: Open firewalld port for etcd embedded in OpenShift
  firewalld: port={{ item[0] }} permanent={{ item[1] }} state=enabled
  with_nested:
    - [ 4001/tcp, 8080/tcp, 8443/tcp, 8444/tcp ]
    - [ true, false ]
```

Port 8444 is used for web console

File: roles/openshift_master/tasks/main.yml [hunk 2]
```diff
@@ -24,10 +24,10 @@
   firewalld: port=4001/tcp permanent=true state=enabled
 
 - name: Open firewalld port for OpenShift
```
Comment (akram) [inline roles/openshift_master/tasks/main.yml]: You could also factorise the declaration of the firewalld in a more concise syntax like this:
```
- name: Open firewalld port for etcd embedded in OpenShift
  firewalld: port={{ item[0] }} permanent={{ item[1] }} state=enabled
  with_nested:
    - [ 4001/tcp, 8080/tcp, 8443/tcp, 8444/tcp ]
    - [ true, false ]
```
Port 8444 is used for web console

Comment (detiber) [inline roles/openshift_master/tasks/main.yml]: @akram Nice, didn't think about using nested there...  I'm looking forward to the immediate flag being introduced with ansible 1.9 to remove the need to issue 2 firewalld statements for each rule

**Comment by jwhonce** [comment]
@detiber Did you want me to add that code in this PR, or are you going to create a PR with it?

**Comment by detiber** [comment]
@jwhonce I can create an additional PR for that. Incoming shortly

File: roles/openshift_node/tasks/main.yml (hunk 1 above)
Comment (twiest) [inline roles/openshift_node/tasks/main.yml]: We've decided to allow this with a FIXME comment until we fix the larger rsync / kubelet restart ordering issue (double configure steps) in a separate PR.

**Pull request merged**